paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/FlinkSourceBuilder.java

```diff
@@ -237,6 +250,20 @@ private TypeInformation<RowData> produceTypeInfo() {
         return InternalTypeInfo.of(produceType);
     }
 
+    private @Nullable org.apache.paimon.types.RowType projectedRowType() {
+        return Optional.ofNullable(projectedFields)
+                .map(Projection::of)
+                .map(p -> p.project(table.rowType()))
+                .orElse(null);
+    }
+
+    private @Nullable ProjectionRowData projectedRowData() {
+        return Optional.ofNullable(projectedFields)
+                .map(Projection::of)
+                .map(p -> p.getRowData(table.rowType()))
```

Comment:
I think it is hard to get something from method name `getRowData`. Do you mean `getOuterProjectRow`?